docs/dev/lsp-extensions.md

```diff
@@ -839,3 +839,26 @@ export interface Diagnostic {
         rendered?: string;
     };
 }
+```
+
+## Dependency Tree
+
+**Method:** `rust-analyzer/fetchDependencyGraph`
+
+**Request:**
+
+```typescript
+export interface FetchDependencyGraphParams {}
+```
+
+**Response:** 
+```typescript
+export interface FetchDependencyGraphResult {
+    crates: {
+        name: string;
+        version: string;
+        path: string;
```

Comment:
Wouldn't it be more general (and more in line with the language server protocol) to use `textDocument: TextDocumentIdentifier` here instead of the current `path: string`?

And as a side note: why this is called a graph/tree, when this is just a list?  It is not possible to construct a graph from the response.  Ie.: visualize the the dependencies of a dependency.

Reply:
@nemethf you're right about the naming, I changed from "graph" to "list". The tree is because the crates are displayed in a "ViewTree".
About the `TextDocumentIdentifier `, I tried using it, but it just made the code more complicated, because to reveal the items in the viewTree I need to work with the filesystem path, and the `TextDocumentIdentifier` inputs the "scheme" in the path.

Reply:
> The tree is because the crates are displayed in a "ViewTree". 

Thank you.  I have no objections against it.  However, it is a bit strange that a client-side implementation detail determines an abstract name in protocol specification.  I can imagine some clients using this extension for something else, for example, to provide a "jump to dependency" functionality that implements an incremental search on the dependencies without showing them beforehand. 

Also this fetchDependencyGraph reminds me of the rust-analyzer/viewCrateGraph LSP extension.  In the long run, it might make sense to generalize both of them to send the same details.   fetchDependencyGraph could list the dependencies of a dependency, and viewCrateGraph could be extended to have a uri/path for each dependency.

Reply:
A `TextDocumentIdentifier` would indeed make more sense if possible